eventbus/driver/nats.go

```diff
@@ -202,14 +239,22 @@ func (n *natsStreaming) processEventSourceMsg(m *stan.Msg, msgHolder *eventSourc
 		}
 	}
 
+	// NATS Streaming guarantees At Least Once delivery,
+	// so need to check if the message is duplicate
+	if _, ok := msgHolder.smap.Load(event.ID()); ok {
+		log.Infow("ATTENTION: Duplicate delivered message detected", "message", m)
+		_ = m.Ack()
```

Comment:
don't think this should be an ack in case the other receive doesn't end up acking

Reply:
So why reack the msg then?

Reply:
The way recommended by NATS maintainers is "if you don't want the event, simply ack it". The `ack` here only means `ok, I know.` for NATS streaming server.

Reply:
i'm wondering though if it wasn't acked and the map has the id that could mean the body exited before the action was triggered so it wasn't actually processed. In that case we'd probably want the ack timeout to play out in order to retry

Reply:
I don't quite understand that. To my opinion, if the ID is in the map, means it has been acked. In some cases, ack will trigger an action - when there's only one dependency event source; In some other cases, ack will not trigger an action - i.e. there are 2 dependencies, dep1 && dep2, dep1 has 2 messages already, but still waiting for dep2, then the first message in dep1 needed to be acked, but there's no action will be triggered. This is by design - we only take the last message for one dependency.